blueprints/hr_assistant/date.py

```diff
@@ -0,0 +1,343 @@
+        elif len(time_ent[i].split('-')) == 3:
+            continue
+
+        # resolve years entities (eg. 2016) and capture the whole year as duration
+        elif len(re.split('-|\\|/', time_ent[i])) == 1:
+
+            # check if time entity is numeric
+            try:
+                int(time_ent[i])
+            except ValueError:
+                return
+
+            # add duration (i.e. start and end dates) to the time entities list
+            try:
+                # Consider the beginning of the year as the comparator point if
+                # date compare entity is provided
+                if date_compare_ent:
+                    time_ent[i] = str(time_ent[i])+'-01-01'
+
+                # If no date compare entity, add both beginning and end of year
+                # dates to the time entities (to keep track of the whole duration)
+                else:
+                    # time_old : year mentioned
+                    # Added entities: 01-01-YEAR and 12-31-YEAR to cover duration
+                    time_old = str(time_ent[i])
+                    time_ent[i] = time_old+'-01-01'
+                    time_ent.append(time_old+'-12-31')
+            except Exception:
+                return
+
+        else:
+            return
+
+    return time_ent
+
+
+def _resolve_time(request, responder, qa, size):
+    """
+    This helper function is useful to resolve any time related entities in the database, including
+    both system entities and custom entities. It captures the type of date entity being talked about
+    by the user (say employment related -- hired/fired, or birth). It in-turn determines the time
+    period being talked about and filters the knowledge to narrow it down to only the datapoints
+    satisfying the duration. The returned value is this shortlisted set of employees.
+    """
+
+    # 'action' entities represent employment action such as hiring of termination
+    # 'dob' entities represent the date of birth entity
+
+    # Fetch existing action entity from a previous turn and clear 'action' context
+    action_entity = []
+    if request.frame.get('action'):
+        action_entity.append(request.frame.get('action'))
+        responder.frame['action'] = None
+
+    time_ent = [e['text'] for e in request.entities if e['type'] == 'sys_time']
+    date_compare_ent = [e for e in request.entities if e['type'] == 'date_compare']
+
+    # Catch new action entity and update the existing one in context (if any)
+    new_action_entity = [e['value'][0]['cname']
+                         for e in request.entities
+                         if e['type'] == 'employment_action']
+
+    dob_entity = [e for e in request.entities if e['type'] == 'dob']
+
+    if new_action_entity:
+        action_entity = [e['value'][0]['cname']
+                         for e in request.entities
+                         if e['type'] == 'employment_action']
+
+    actions = {'hired': 'doh', 'fired': 'dot'}
```

Comment:
These dictionaries need to be made into a constant that are accessible across multiple files.